packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -70,6 +70,10 @@ const Widget _startSelectionOverlay = CupertinoPickerDefaultSelectionOverlay(cap
 const Widget _centerSelectionOverlay = CupertinoPickerDefaultSelectionOverlay(capStartEdge: false, capEndEdge: false);
 const Widget _endSelectionOverlay = CupertinoPickerDefaultSelectionOverlay(capStartEdge: false);
 
+/// Defines a function signature for creating a widget that serves as a selection overlay,
+/// given the current context, the selected item's index, and the total number of columns.
+typedef SelectionOverlayBuilder = Widget? Function(BuildContext context, int index, int columnCount);
```

Comment:
Super nit: Maybe rename to `selectedIndex` and put it last, after `columnCount`, since it is an index in `columnCount`.

Reply:
Another nit: You could also consider making these named parameters. I see below where it is called, it's a little confusing seeing `(context, 2, 3)` etc. and not remembering exactly what the numbers mean.